docs/rfc/rfc-1.adoc

```diff
@@ -458,11 +460,31 @@ To make sure that older wallets are still accessible for redemption, we need to
 perform heartbeats. The signing group signs a block when block count mod
 `heartbeat_block_length` = 0 and then does _not_ publish the result. If there
 are ever less than `heartbeat` operators that participate in the heartbeat, the
-ones that did can create and sign a new transaction that lists the _inactive_
-operators. Once this transaction is posted to the ethereum chain, we can
-iterate through the inactive operators, remove them from reward eligibility
-for `failed_heartbeat_reward_removal_period` amount of time, transfer the remaining
-BTC from the wallet to another random wallet and close this wallet.
+active operators record the inactive operators. If this happens a number of
+times >= `consecutive_failed_heartbeats`, the operators take a union the
+recorded _inactive_ operators, and an operator from among the active operators
+posts a transaction to disable those inactive operators from receiving rewards
+for `failed_heartbeat_reward_removal_period` amount of time. The active
+operators move the remaining BTC to another random wallet and close this
+wallet.
+
+For the purposes of heartbeats, an operator that is currently unstaking (they
+started their two-week undelegation period) does not count as a live heartbeat,
+but also does not count as an inactive operator. Thus, operators who are
+unstaking might cause a wallet to fail a heartbeat, but can still be around to
+help move the funds to another wallet, and would not have their rewards turned
+off.
```

Comment:
Unstaking operators are earning rewards as long as they do not fail a heartbeat in one of their wallets. Does it not incentivize them to modify clients to participate in heartbeats? 🤔

Reply:
Since the heartbeat happens off-chain, and whether or not you have unstaked is public information (i think), each operator can verify that the message sent from each other operator is truthful off-chain.

For example, say that I unstake, but then modify my client to still send heartbeats. You should be able to check my staking status and see that my heartbeat was a fake heartbeat and discard it.

If those assumptions are true I'll stick that in the RFC

Reply:
We will have to cache unstaking operators in-memory, observe events on the token staking contract and add operators to the cache every time we see an unstake event. Othwerise, every heartbeat verification would get really expensive for operator running with Alchemy/Infura.

But yeah, I think it's a good direction and what I am saying here is just an implementation detail.

Reply:
Are you okay with that added complexity? I'm drawing a blank for other ways to make it work:

* We want the folks who are unstaking to stick around to help transfer funds
* We don't want them to modify their clients
* We want unstaking to create heartbeat pressure (otherwise when they're gone we suddenly can't move funds)

Reply:
I think it makes sense. I am wondering if we should expect unstaking operators to participate in the heartbeat but filter them off by other clients when determining if we are still above the threshold. Then, even if the wallet fails the threshold, it does not mark unstaking but active operators as non-eligible for rewards.

This way:
- unstaking operators who are still around to sign redemption requests receive rewards,
- unstaking operators who turned off their nodes are treated just like non-unstaking operators who turned off their nodes and do not receive rewards.

What do you think?

Reply:
Love it!

Reply:
addressed in 0a0363c